common/lib/dependabot/pull_request_creator/github.rb

```diff
@@ -110,6 +110,13 @@ def initialize(source:, branch_name:, base_commit:, credentials:,
 
       sig { returns(T.untyped) }
       def create
+        if trace_log? && branch_exists?(branch_name)
+          Dependabot.logger.info(
+            "Existing branch \"#{branch_name}\" found: raising DuplicateBranchExists exception."
+          )
+          raise DuplicateBranchExists, "Duplicate branch #{branch_name} already exists"
+        end
+
```

Comment:
If we are raising when a branch already exists then we can get rid of the checks for when an unmerged pull request exists. But it might be better to do that in a follow-up PR since this one is already a lot of changes

Reply:
@Nishnha , ill do it next release